prometheus/vec.go

```diff
@@ -216,7 +228,7 @@ func (m *MetricVec) hashLabelValues(vals []string) (uint64, error) {
 	}
 	h := hashNew()
 	for _, val := range vals {
-		h = hashAdd(h, val)
+		h = m.hashAdd(h, val)
```

Comment:
> The 0xFF character can never be part of a valid UTF-8 string. That qualifies it as a good separator character during hashing.

I don't see this link but I've used the constant `model.SeparatorByte`.

To give you context, we had a [sort error](https://github.com/docker/distribution/pull/1888) in the docker registry API due to the use of a `/` that sorts earlier than a `-`. Changing to `0x00` gives you expected, lexical sort order.